hyperdome/client/api.py

```diff
@@ -21,39 +21,46 @@
 
 import functools
 import json
-import typing
-
+from typing import ParamSpec, Callable, Concatenate
 import autologging
-import requests
+from PyQt5 import QtWebSockets
+from PyQt5.QtNetwork import (
+    QNetworkAccessManager,
+    QNetworkProxy,
+    QNetworkReply,
+    QNetworkRequest,
+)
+from PyQt5.QtCore import QUrl, pyqtSlot
 
 from ..common.server import Server
 
 
-def handle_requests_errors(fn: typing.Callable):
-    """
-    wraps a requests call to provide exception handling boilerplate
-    for errors which are generic to any requests call
-    """
+FnParams = ParamSpec("FnParams")
+CallbackParams = ParamSpec("CallbackParams")
 
-    @functools.wraps(fn)
-    @autologging.logged
-    def wrapper(*args, **kwargs):
-        try:
-            response = fn(*args, **kwargs)
-        except requests.ConnectionError:
-            wrapper._log.warning("couldn't connect to server")
-            raise
-        except requests.Timeout:
-            wrapper._log.warning("server timed out during request")
-            raise
-        except requests.HTTPError as e:
-            wrapper._log.warning(f"{e.args[0]}")
-            raise
-        except:
-            wrapper._log.exception("unexpected exception during request handling")
-            raise
-        else:
-            return response
+
+def attach_callback(
+    fn: Callable[Concatenate[Callable[CallbackParams, None], FnParams], None],
+    *args: FnParams.args,
+    **kwargs: FnParams.kwargs,
+) -> Callable[[Callable[CallbackParams, None]], None]:
+    def decorate(callback: Callable[CallbackParams, None]) -> None:
+        return fn(callback, *args, **kwargs)
+
+    return decorate
+
+
+def error_handler(err: QNetworkReply.NetworkError):
+    if err == err.NoError:
+        return
```

Comment:
Should there be another line after this?

Reply:
Function is unused actually, I should delete it. There's a different way to do error handling that makes more sense for this Qt business